gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -160,6 +160,11 @@ func (vm *VMKeeper) AddPackage(ctx sdk.Context, msg MsgAddPackage) error {
 		return ErrInvalidPkgPath("reserved package name: " + pkgPath)
 	}
 
+	// Validate Gno syntax and type check.
+	if err := gno.TypeCheckMemPackage(memPkg, store); err != nil {
```

Comment:
It's a good choice to put this kind of validation here, because you want to force the chain keeper to validate the message (outside of cookie cutter verifications) the moment it's parsing it